shared_model/backend/protobuf/queries/proto_blocks_query.hpp

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_SHARED_MODEL_PROTO_BLOCKS_QUERY_HPP
+#define IROHA_SHARED_MODEL_PROTO_BLOCKS_QUERY_HPP
+
+#include "backend/protobuf/common_objects/signature.hpp"
+#include "interfaces/queries/blocks_query.hpp"
+#include "utils/lazy_initializer.hpp"
+
+namespace shared_model {
+  namespace proto {
+    class BlocksQuery FINAL : public CopyableProto<interface::BlocksQuery,
+                                                   iroha::protocol::BlocksQuery,
+                                                   BlocksQuery> {
+     private:
+      template <typename T>
+      using Lazy = detail::LazyInitializer<T>;
+     public:
+      template <typename BlocksQueryType>
+      explicit BlocksQuery(BlocksQueryType &&query)
+          : CopyableProto(std::forward<BlocksQueryType>(query)) {}
+
+      BlocksQuery(const BlocksQuery &o) : BlocksQuery(o.proto_) {}
```

Comment:
Not needed, overshadowed by universal reference, same for move constructor :(

Reply:
They are necessary. Otherwise, it isn't compilable

Reply:
Yeah you are right.